packages/common/src/downloads/helpers/file-url-fetchers/fetchExportItemDownloadFileUrl.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  IExportItemRequestOptions,
+  IExportParameters,
+  exportItem,
+  getItemStatus,
+} from "@esri/arcgis-rest-portal";
+import {
+  DownloadOperationStatus,
+  ExportItemFormat,
+  IFetchDownloadFileUrlOptions,
+  LegacyExportItemFormat,
+  PORTAL_EXPORT_TYPES,
+  ServiceDownloadFormat,
+  downloadProgressCallback,
+} from "../types";
+import { getExportItemDataUrl } from "../getExportItemDataUrl";
+import HubError from "../../../HubError";
+import { IArcGISContext } from "../../../ArcGISContext";
+
+export async function fetchExportItemDownloadFileUrl(
+  options: IFetchDownloadFileUrlOptions
+): Promise<string> {
+  validateOptions(options);
+  const { entity, format, context, progressCallback } = options;
+  progressCallback && progressCallback(DownloadOperationStatus.PENDING);
+  const { exportItemId, jobId } = await exportItem({
+    id: entity.id,
+    exportFormat: getExportFormatParam(format as ExportItemFormat),
+    exportParameters: getExportParameters(options),
+    authentication: context.hubRequestOptions.authentication,
+  });
+
+  await pollForJobCompletion(exportItemId, jobId, context);
+
+  // TODO: Once the job is completed, we still need to set the special typekeywords needed to find the item later.
+  // Also, I _think_ we can only do one layer at a time (at least with the current typeKeywords schema we're using)
+  progressCallback && progressCallback(DownloadOperationStatus.COMPLETED);
+  return getExportItemDataUrl(exportItemId, context);
+}
+
+function validateOptions(options: IFetchDownloadFileUrlOptions) {
+  const { geometry, where } = options;
+
+  if (geometry) {
+    throw new HubError(
+      "fetchExportItemDownloadFileUrl",
+      "Geometric filters are not supported for this type of download"
+    );
+  }
+
+  if (where) {
+    throw new HubError(
+      "fetchExportItemDownloadFileUrl",
+      "Attribute filters are not supported for this type of download"
+    );
+  }
+}
+
+function getExportFormatParam(
+  format: ExportItemFormat
+): IExportItemRequestOptions["exportFormat"] {
+  const legacyFormat = getLegacyExportItemFormat(format);
+  return PORTAL_EXPORT_TYPES[legacyFormat]
+    ?.name as IExportItemRequestOptions["exportFormat"];
+}
+
+function getLegacyExportItemFormat(
+  format: ExportItemFormat
+): LegacyExportItemFormat {
+  return format === ServiceDownloadFormat.FILE_GDB ? "fileGeodatabase" : format;
+}
+
+function getExportParameters(
+  options: IFetchDownloadFileUrlOptions
+): IExportParameters {
+  const { layers, where } = options;
+  const result: IExportParameters = {};
+
+  if (layers) {
+    result.layers = layers.map((id) => ({ id, where }));
+  }
+
+  return result;
+}
+
+async function pollForJobCompletion(
```

Comment:
seems to me like this is a good use case for that. I feel like download times increase exponentially w/ the size of the data.

Reply:
Good callout! I'd be curious to hear what product thinks. Since we have to display a progress indicator with a percentage, I'd wager we want a consistent timeout. I suspect that the exponential timeouts would cause users to think that the download operation has frozen (particularly for larger downloads)